clang/lib/Sema/SemaType.cpp

```diff
@@ -7024,13 +7024,12 @@ static bool handleFunctionTypeAttr(TypeProcessingState &state, ParsedAttr &attr,
                << (int)Sema::CallingConventionIgnoredReason::VariadicFunction;
 
       attr.setInvalid();
-      if (S.getLangOpts().SYCLIsDevice) {
-        S.SYCLDiagIfDeviceCode(attr.getLoc(), diag::err_cconv_varargs)
-          << FunctionType::getNameForCallConv(CC);
-        return true;
-      } else
-      return S.Diag(attr.getLoc(), diag::err_cconv_varargs)
-             << FunctionType::getNameForCallConv(CC);
+      // Disable this diagnostic for SYCL
+      // Note that deferring does not work since the declaration
+      // need not be in a function context.
+      if (!S.getLangOpts().SYCLIsDevice)
```

Comment:
I think we just need to remove all SYCL specific code from this part as varargs function is detected by SemaSYCL checks.
Could you clarify why this check is still needed? 
Is the problem is `spir_function` calling convention or `__cdecl`?

Reply:
You are right, this is no longer needed.   I have remove the SYCL-specific changes.